src/zarr/storage/logging.py

```diff
@@ -203,19 +205,19 @@ async def set_partial_values(
         with self.log(keys):
             return await self._store.set_partial_values(key_start_values=key_start_values)
 
-    async def list(self) -> AsyncIterator[str]:
+    async def list(self) -> AsyncGenerator[str, None]:
```

Comment:
What's the reason for switching these back to `AsyncGenerators`? 

xref: https://github.com/zarr-developers/zarr-python/pull/2381

Reply:
this was likely me blindly obeying a wayward linter. I will revert